### PR TITLE
feat: UI fixes, persist messages, set up cxn limit

### PR DESCRIPTION
**Pull request by maschad**
This PR addresses a few things:

- [x]  remove Peer ID to connect to and multiaddr to connect to
- [x]  persist chat messages on chat tab (when switching between chat tab and connecting to peer tab, the chat messages are lost)
- [x]  set up max connection limit in js-libp2p-gossipsub (limit the size of meshes) 
- [x] Refactor UI including Icons and burger button

image

image

Related #35

### CODE REVIEW

File: packages/frontend/src/lib/libp2p.ts [hunk 1]
```diff
@@ -1,43 +1,32 @@
...
 import type { Message, SignedMessage } from '@libp2p/interface-pubsub'
-import { LevelDatastore } from 'datastore-level'
-import isIPPrivate from 'private-ip'
 import { gossipsub } from '@chainsafe/libp2p-gossipsub'
 import { webSockets } from '@libp2p/websockets'
 import { webTransport } from '@libp2p/webtransport'
 import { webRTC, webRTCDirect } from '@libp2p/webrtc'
-import { CHAT_TOPIC, CIRCUIT_RELAY_CODE } from './constants'
+import { BOOTSTRAP_NODE, CHAT_TOPIC, CIRCUIT_RELAY_CODE } from './constants'
 import * as filters from "@libp2p/websockets/filters"
```
Comment (p-shahi) [inline packages/frontend/src/lib/libp2p.ts:17]: I think we can remove this too

File: rust-peer/src/main.rs [hunk 2]
```diff
@@ -169,10 +176,7 @@ async fn main() -> Result<()> {
                     debug!("Failed to run Kademlia bootstrap: {e:?}");
                 }
 
-                let message = format!(
-                    "Hello world! Sent from the rust-peer at: {:4}s",
-                    now.elapsed().as_secs_f64()
-                );
+                let message = format!("My social skills are a little rusty...");
```
Comment (p-shahi) [inline rust-peer/src/main.rs:179]: Let's revert this

File: rust-peer/src/main.rs [hunk 3]
```diff
@@ -191,11 +195,14 @@ struct Behaviour {
     identify: identify::Behaviour,
     kademlia: Kademlia<MemoryStore>,
     keep_alive: keep_alive::Behaviour,
-    // ping: ping::Behaviour,
+    ping: ping::Behaviour,
```
Comment (p-shahi) [inline rust-peer/src/main.rs:198]: We don't need ping imo. Was this for debugging?

Comment (maschad) [reply rust-peer/src/main.rs:198]: it's necessary if we want to display the [Latency of nearest peer](https://github.com/libp2p/universal-connectivity/pull/39/files#diff-2ddf273f11d6f40730b230314e354e9b900ee9a6190738168c8d8d18b7a369b4R96)

Comment (p-shahi) [reply rust-peer/src/main.rs:198]: I say remove. The latency is interesting but not at the cost the of the inbound connections issue

**Review by p-shahi — APPROVED** [review]
Awesome thank you so much

**Comment by thomaseizinger** [comment]
Nice work!

Perhaps the first page could still have a few things renamed, now that it is completely passive.

I'll send a PR.

**Comment by DougAnderson444** [comment]
I kind of liked being able to add a peerID/ MultiAddr on that first tab, what was the rationale behind removing it?

**Comment by maschad** [comment]
> I kind of liked being able to add a peerID/ MultiAddr on that first tab, what was the rationale behind removing it?

We now bootstrap to the deployed rust peer then dial the peers once they get added to the DHT

**Comment by DougAnderson444** [comment]
Ok perfect, I added a note in the README in my PR on how to connect to that via cli arg.

**Pull request merged**